openadapt/utils.py

```diff
@@ -442,35 +445,42 @@ def take_screenshot() -> Image.Image:
     Returns:
         PIL.Image.Image: The screenshot image.
     """
-    with mss.mss() as sct:
-        monitors = sct.monitors
-
-        if config.CAPTURE_ALL_MONITORS:
-            # Determine the bounds of the combined image
-            min_left = min(monitor['left'] for monitor in monitors[1:])
-            min_top = min(monitor['top'] for monitor in monitors[1:])
-            max_right = max(monitor['left'] + monitor['width'] for monitor in monitors[1:])
-            max_bottom = max(monitor['top'] + monitor['height'] for monitor in monitors[1:])
-
-            total_width = max_right - min_left
-            total_height = max_bottom - min_top
-            
-            combined_image = Image.new("RGB", (total_width, total_height))
-
-            for monitor in monitors[1:]:  # Skip the first entry which is a union of all monitors
-                sct_img = sct.grab(monitor)
-                image = Image.frombytes("RGB", sct_img.size, sct_img.bgra, "raw", "BGRX")
-                x_offset = monitor['left'] - min_left
-                y_offset = monitor['top'] - min_top
-                combined_image.paste(image, (x_offset, y_offset))
-            
-            return combined_image
-        else:
-            # Capture the current monitor
-            current_monitor = get_current_monitor(monitors)
-            sct_img = sct.grab(current_monitor)
-            image = Image.frombytes("RGB", sct_img.size, sct_img.bgra, "raw", "BGRX")
-            return image
+    global SCT
+    monitors = SCT.monitors[1:]  # Skip the first entry which is a union of all monitors
+
+    if config.CAPTURE_ALL_MONITORS:
+        # Grab all monitors at once
+        sct_img = SCT.grab(SCT.monitors[0])  # Grab the union of all monitors
+        full_img = Image.frombytes("RGB", sct_img.size, sct_img.bgra, "raw", "BGRX")
```

Comment:
@onyedikachi-david can you please clarify why the rest of this block is necessary? Why not just return `full_img` directly?